compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SemanticAnalyzer.java

```diff
@@ -2350,6 +2350,10 @@ public void visit(BLangIf ifNode) {
 
         Map<BVarSymbol, BType.NarrowedTypes> prevNarrowedTypeInfo = this.narrowedTypeInfo;
 
+        // This map keeps the narrowed types of inner if statements and propagate the false types to else block when
+        // flow goes from the if statement to else block in compile time.
```

Comment:
Isn't it for after the else block completes rather than for the else block?

Reply:
I changed the comment, saying that we want to keep track of the false types after the else blocks, which we might need in the outer block